rust/connlib/tunnel/src/client.rs

```diff
@@ -644,6 +641,21 @@ impl IpProvider {
             ipv6: Box::new(ipv6.subnets_with_prefix(128).map(|ip| ip.network_address())),
         }
     }
+
+    pub fn get_ip_for(&mut self, ip: &IpAddr) -> Option<IpAddr> {
+        let ip = match ip {
+            IpAddr::V4(_) => self.ipv4.next().map(Into::into),
+            IpAddr::V6(_) => self.ipv6.next().map(Into::into),
+        };
+
+        if ip.is_none() {
+            // TODO: we might want to make the iterator cyclic or another strategy to prevent ip exhaustion
```

Comment:
When would this happen? If there are more than 2 million matched domains?

Reply:
Exactly